internal/controller/controllers/agentserviceconfig_controller_test.go

```diff
@@ -1682,27 +1682,95 @@ var _ = Describe("Default ConfigMap values", func() {
-		configMap = cm.(*corev1.ConfigMap)
-	})
+	Context("Default values", func() {
+		BeforeEach(func() {
+			var ASCC ASC
+			asc := newASCDefault()
+			ascr := newTestReconciler(asc, route, imageRoute)
+			ASCC.init(ascr, asc)
+			cm, mutateFn, err := newAssistedCM(ctx, log, ASCC)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(mutateFn()).ShouldNot(HaveOccurred())
+			configMap = cm.(*corev1.ConfigMap)
+		})
+		It("INSTALL_INVOKER", func() {
+			Expect(configMap.Data["INSTALL_INVOKER"]).To(Equal("assisted-installer-operator"))
+		})
 
-	It("INSTALL_INVOKER", func() {
-		Expect(configMap.Data["INSTALL_INVOKER"]).To(Equal("assisted-installer-operator"))
+		It("sets the base URLs", func() {
+			Expect(configMap.Data["SERVICE_BASE_URL"]).To(Equal(fmt.Sprintf("https://%s", testHost)))
+			Expect(configMap.Data["IMAGE_SERVICE_BASE_URL"]).To(Equal(fmt.Sprintf("https://%s.images", testHost)))
+		})
 	})
 
-	It("sets the base URLs", func() {
-		Expect(configMap.Data["SERVICE_BASE_URL"]).To(Equal(fmt.Sprintf("https://%s", testHost)))
-		Expect(configMap.Data["IMAGE_SERVICE_BASE_URL"]).To(Equal(fmt.Sprintf("https://%s.images", testHost)))
+	Context("Public Container Registries", func() {
+		var (
+			registryConf = `
+	unqualified-search-registries = ["registry.access.redhat.com","docker.io"]
+	[[registry]]
+		prefix = ""
+		location = "quay.io/edge-infrastructure"
+		mirror-by-digest-only = false
+
+	[[registry.mirror]]
+		location = "mirror1.registry.corp.com:5000/edge-infrastructure"`
+			mirrorCM = &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testMirrorRegConfigmapName,
+					Namespace: testNamespace,
+				},
+				Data: map[string]string{
+					mirrorRegistryRefCertKey: "foo",
+				},
+			}
+			asc = newASCDefault()
+		)
+
+		It("default public container registries", func() {
+			ascr := newTestReconciler(asc, route, imageRoute)
+			checkPublicContainerRegistries(ctx, log, ascr, asc, "quay.io,registry.svc.ci.openshift.org")
+		})
+		It("adds mirror registries", func() {
+			asc.Spec.MirrorRegistryRef = &corev1.LocalObjectReference{Name: testMirrorRegConfigmapName}
+			mirrorCM.Data[mirrorRegistryRefRegistryConfKey] = registryConf
+			ascr := newTestReconciler(asc, route, imageRoute, mirrorCM)
+			checkPublicContainerRegistries(ctx, log, ascr, asc, "quay.io,registry.svc.ci.openshift.org,registry.access.redhat.com,docker.io")
+		})
+		It("adds user-specified public container registries", func() {
+			asc.Spec.PublicContainerRegistries = []string{"example.com"}
+			ascr := newTestReconciler(asc, route, imageRoute)
+			checkPublicContainerRegistries(ctx, log, ascr, asc, "quay.io,registry.svc.ci.openshift.org,example.com")
+		})
+		It("adds user-specified public container registries", func() {
+			asc.Spec.PublicContainerRegistries = []string{"example.com"}
+			ascr := newTestReconciler(asc, route, imageRoute)
+			checkPublicContainerRegistries(ctx, log, ascr, asc, "quay.io,registry.svc.ci.openshift.org,example.com")
+		})
```

Comment:
Are these tests the same?

Reply:
Yes 😆 I was fixing the generated files and this must've been a merge error. Thanks for catching! Removed the duplicate